Makefile

```diff
@@ -1,6 +1,22 @@
-all:
-	go build GNU_sort.go
-	./GNU_sort GNU_sort.go
+.PHONY: clean
+
+PATH2MAIN := "./cmd/..."
+
+all: build
 
 clean:
-	rm GNU_sort
\ No newline at end of file
+	rm -f sort *.out
+
+build:	
+	go build $(PATH2MAIN)
+
+tests: build
+	./test/test_options.sh
+	./test/test_diff_1.sh
+	./test/test_diff_2.sh
+	./test/test_diff_3.sh
+	./test/test_long.sh
+	./test/test_multiple.sh
+	./test/test_reverse.sh
+	cd internal/sort && go test
```

Comment:
Nothing weird here but just for the sake of your own knowledge: [pushd](https://www.gnu.org/software/bash/manual/bash.html#index-pushd) and [popd](https://www.gnu.org/software/bash/manual/bash.html#index-popd)

Reply:
Okay, did not know about these